File: linda-presentation/src/main/java/it/unibo/coordination/linda/presentation/RegexTemplateSerializer.java
package it.unibo.coordination.linda.presentation;

import com.fasterxml.jackson.databind.ObjectMapper;
import it.unibo.coordination.linda.text.RegexTemplate;

import static it.unibo.coordination.utils.CollectionUtils.mapOf;

class RegexTemplateSerializer extends DynamicSerializer<RegexTemplate> {

    public RegexTemplateSerializer(MIMETypes mimeType, ObjectMapper mapper) {
        super(mimeType, mapper);
    }

    @Override
    public Object toDynamicObject(RegexTemplate object) {
        return mapOf("template", object.getTemplate().pattern());
    }
}

File: linda-text-presentation/src/main/kotlin/it/unibo/coordination/linda/text/RegexTemplateSerializer.kt
package it.unibo.coordination.linda.text

import com.fasterxml.jackson.databind.ObjectMapper
import it.unibo.presentation.DynamicSerializer
import it.unibo.presentation.MIMETypes

internal class RegexTemplateSerializer(mimeType: MIMETypes, mapper: ObjectMapper) : DynamicSerializer<RegexTemplate>(mimeType, mapper) {
    override fun toDynamicObject(`object`: RegexTemplate): Any {
        return mapOf("template" to `object`.template.pattern())
    }
}